File: python/discount.py
import sdk.python.entities
from sdk.python.entities import Property
from sdk.python.discount_group import DiscountGroup


class Discount(sdk.python.entities.Entity):

    resource = '/discount/'
    json_name = 'discount'

    id = Property(int)
    lower_limit = Property(float)
    amount = Property(float)
    discount_group = Property(DiscountGroup)


class Discounts(sdk.python.entities.Resource):

    entity_class = Discount
    json_name = 'discounts'


discounts = Discounts()

File: javascript/discount.js
import generateUUID from './uuid.js';
import { addPropertyTo } from './model.js';

export function Discount() {
    this.resource = '/discounts';
    this.json = 'discount';
    this.temporaryId = generateUUID();

    addPropertyTo(this, 'id');
    addPropertyTo(this, 'lowerLimit');
    addPropertyTo(this, 'amount');

    this.discountedUnitCost = function (unitPrice) {
        var unitCost = unitPrice ? unitPrice : 0,
            amount = this.amount() ? this.amount() : 0,
            discount = 100 - amount;
        return parseFloat((unitCost / 100) * discount).toFixed(3);
    };
}
